Documented models for atmospheric scattering and absorption

nollety opened the pull request:
# Description

I've added a new section to the user guide to describe what models we use for molecular scattering and molecular absorption in the atmosphere.

# Checklist

- ~~[ ] The code follows the relevant coding guidelines~~
- ~~[ ] The code generates no new warnings~~
- ~~[ ] The code is appropriately documented~~
- ~~[ ] The code is tested to prove its function~~
- [x] The feature branch is rebased on the current state of the `main` branch
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

leroyvn commented on docs/rst/user_guide/atmosphere/absorption.rst:
```diff
@@ -0,0 +1,112 @@
+   k_{\mathrm a \lambda} = n \, \sigma_{\mathrm a}(\nu, p, T)
+
+where
+:math:`n` is the air number density,
+:math:`\sigma` is the monochromatic absorption cross section,
+:math:`\nu = \lambda^{-1}` is the wavenumber,
+:math:`p` is the pressure and
+:math:`T` is the temperature.
```
```suggestion
* :math:`n` is the air number density,
* :math:`\sigma` is the monochromatic absorption cross section,
* :math:`\nu = \lambda^{-1}` is the wavenumber,
* :math:`p` is the pressure and
* :math:`T` is the temperature.
```

leroyvn commented on docs/rst/user_guide/atmosphere/absorption.rst:
```diff
@@ -0,0 +1,112 @@
+     - 0.0314
+     - 0.0002
+
+The ``us76_u86_4`` is named in this way because it corresponds to the
+gas mixture defined by the ``us76`` atmosphere model
+:cite:`NASA1976USStandardAtmosphere` in the region of altitudes below 86 km,
+and restricted to the 4 main molecular species, namely N2, O2, CO2 and CH4.
```
This should go above the table: it describes what this profile is really about.

leroyvn commented on docs/rst/user_guide/intro.rst:
```diff
@@ -8,6 +8,7 @@ User guide
 
    basic_concepts
    onedim_solver_app
+   atmosphere/intro
    package_structure
    data_guide
    unit_guide_user
```
Let's put modelling notes at the end after the content dedicated to using the software.

```suggestion
   basic_concepts
   onedim_solver_app
   package_structure
   data_guide
   unit_guide_user
   atmosphere/intro
```

leroyvn commented on docs/rst/user_guide/atmosphere/intro.rst:
```diff
@@ -0,0 +1,8 @@
+.. _sec-atmosphere-intro:
```
I'd like to change conventions and use `index.rst` as the landing page for each directory.

leroyvn reviewed (approved):
Hi @nollety, thanks for this PR. I requested a few changes, among which one is to rename `intro.rst` to `index.rst` (convention used in websites and better suited in many cases in our docs because the "intro" page actually most often contains nothing.

nollety commented:
@leroyvn I added your changes. I've also added the dimensions of the physical quantities that are introduced, so as to avoid any confusion, the normalisation rule that we use for the phase function, and other minor things.

leroyvn commented:
Good! As usual: rebase + squash and we'll merge this!

nollety commented:
@leroyvn it's ready to be merged :)
